eps/README.md

```diff
@@ -0,0 +1,106 @@
+# Enhancement Proposal (EP)
+
+This document describes the process of submitting an Enhancement Proposal (EP). EPs are a way to propose, communicate,
+and coordinate on new features for the NGINX Kubernetes Gateway. It is based off of
+the [Gateway Enhancement Proposals](https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/overview.md). Their
+purpose is to:
+
+- Make changes and proposals discoverable (current and future).
+- Provide a common place to discuss design, architecture, and impacts of a particular change.
+- Document design ideas, tradeoffs, and decisions for historical reference.
+
+## Process
+
+The diagram below shows the EP process:
+
+```mermaid
+flowchart TD
+    D([Open Discussion]) --> C
+    C([Issue Created]) --> Provisional
+    Provisional -->|EP Doc PR<br />done| Implementable
+    Provisional -->|If practical <br /> work needed| Prototyping
+    Prototyping -->|EP Doc PR<br />done| Implementable
+    Implementable -->|Work completed| Completed
+```
+
+### 1. Open a GitHub Discussion
+
+Before creating an issue or EP, open
+a [GitHub discussion](https://github.com/nginxinc/nginx-kubernetes-gateway/discussions). Describe the feature you would
+like, any use cases you have, and other relevant details. Beginning with a discussion allows you to get feedback from
+the maintainers and the community before you invest time in writing an EP.
+
+### 2. Create an Issue
+
+If there is consensus on the discussion post that the feature is important and should be included in the roadmap, a
+maintainer will ask you
+to [open an issue](https://github.com/nginxinc/nginx-kubernetes-gateway/issues/new?assignees=&labels=proposal&projects=&template=enhancement.md&title=)
+on GitHub.
+
+### 3. Agree on the Goals (Provisional)
+
```

Comment:
Does this happen within the Issue?  Or is this a PR and a requirement for a new "EP" document?

Reply:
I ask because there might be well understood or small changes that could be captured in an Issue as opposed to adding the additional friction of requiring an EP for everything.
Essentially, I see an EP as a way to describe implementation and get everyone on board with implementation.  Specifically, when that implementation is complex or complicated.

Reply:
> Does this happen within the Issue? Or is this a PR and a requirement for a new "EP" document?
In sig-gateway this is a PR. If we prefer a more lightweight approach, we can change this. 

> I ask because there might be well understood or small changes that could be captured in an Issue as opposed to adding the additional friction of requiring an EP for everything.

I agree. The "Out of scope" section covers the types of issues that don't need an EP. https://github.com/nginxinc/nginx-kubernetes-gateway/blob/f0def492190e75e22083875d7828944962fd1b23/eps/README.md#out-of-scope. The enhancement proposal issue template also has a statement that says "depending on the scope of the enhancement, you make be asked to use the EP process...". 

My thinking is that by default, users will create issues when they want a feature. If the feature requires more discussion, we will ask them to open a discussion. At that point, if it requires an EP, we will point them to this document.